terracotta/image.py

```diff
@@ -187,6 +188,21 @@ def gamma_correction(
     return arr
 
 
+def apply_color_transform(
+        masked_data: Array,
+        color_transform: str,
+        out_dtype: type = np.uint16,
+) -> Array:
+    """Apply gamma correction to the input array and scale it to the output dtype."""
+    arr = to_math_type(masked_data)
+
+    for func in parse_operations(color_transform):
+        arr = func(arr)
```

Comment:
Does this assume a particular scaling already?

Reply:
Yes, inputs have to be in the [0,1] range: https://github.com/vincentsarago/color-operations/?tab=readme-ov-file#color_operationsoperations

Reply:
I see. It may be easier to apply stretching before color transform then. I.e., normalize to `[0,1]` using `stretch_range` and clamp spillover to `0` / `1`, then apply color transform, then convert to `uint8`. That way we don't rely on a transformation of the stretch range?

Reply:
I think it may not be a good idea since it would change the result. 

The color stretch fills the [0, 1] range whereas the normalization `to_math_type` does is relative to the dtype range. So when you apply the color transform I think that would shift where values fall along the curve. 

This is what chatgpt said in the case of gamma correction:
```
Gamma correction is typically applied before color stretching, particularly when working with image data in scientific imaging, photography, or graphic design.

Here's why:

    Gamma Correction First: Gamma correction adjusts the image data to a linear color space, which compensates for the nonlinear response of human vision and many display systems. This linearization step ensures that subsequent adjustments, like color stretching, are applied to data that more accurately represents real-world light intensities.
```

Reply:
Sure, but who says that the dtype range is appropriate? The user-provided stretch range is telling us how to map the values in the raster to a linear scale.

Summoning @vincentsarago in case you want to weigh in on the appropriate order of linear scaling (and clamping out of range values) vs. color correction :)